redgenes/gene_annotations.py

```diff
@@ -227,11 +232,33 @@ def annotation_pipeline(df, tmpdir, kofamscan_profile, kofamscan_kolist):
             args_md_info = entity_id + args_md_info
             TRN.add(sql_md_info, args_md_info)
 
+            prodigal_version = cds_df["source"].tolist()[0]
+            sql_extract_prodigal_run_info = """
+                SELECT run_id FROM run_info
+                WHERE software = 'prodigal' and version = ? and commands = ?"""
+            TRN.add(
+                sql_extract_prodigal_run_info, [prodigal_version, prodigal_commands]
+            )
+            prodigal_run_id = TRN.execute_fetchflatten()
+
+            if not prodigal_run_id:
+                sql_run_info_prodigal = """
+                    INSERT INTO run_info (software, version, commands)
+                    VALUES (?, ?, ?) RETURNING run_id"""
+                args_run_info_prodigal = [
+                    "prodigal",
+                    prodigal_version,
+                    prodigal_commands,
+                ]
+                TRN.add(sql_run_info_prodigal, args_run_info_prodigal)
+                prodigal_run_id = TRN.execute_fetchflatten()
+
             cds_df.insert(0, "entity_id", entity_id[0])
+            cds_df["run_id"] = prodigal_run_id[0]
```

Comment:
Is indexing `prodigal_run_id` always valid? I would assume its a value, rather than a container like a list, as it's described as an ID